### PR TITLE
zapTypeToDecodableClusterObjectType does not throw on ambiguous types

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

When using the `zapTypeToDecodableClusterObjectType` helper, the type may be ambiguous and we will return the first type that match.

#### Change overview
 * Update `zapTypeToDecodableClusterObjectType` to throw if there is some ambiguity 
 * Update `commissioning.xml` by replacing `SecurityType` with `WiFiSecurity` as named in the spec (it was causing some ambiguity)
 * Update `SoftwareFault` event name to `SoftwareFaultEvent` as the name is colliding with the `SoftwareFault` struct which is also named `SoftwareFault`

#### Testing
 Running codegen does not produce ambiguous results anymore.

### CODE REVIEW

**Comment by vivien-apple** [comment]
I have renamed the `SoftwareFault` event to `SoftwareFaultEvent` because the same name is used for the `SofwareFault` struct. But looking at the spec, it is unclear to me what the `SoftwareFault` struct is not inlined in the `SoftwareEvent` itself ?

**Comment by bzbarsky-apple** [comment]
> But looking at the spec, it is unclear to me what the SoftwareFault struct is not inlined in the SoftwareEvent itself ?

As in, why the struct exists as as separate entity?  I think it's just a bug.  This event is not following the DiscoBall event template at all...

I think we should just eliminate this struct, and I filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/4831 on the spec, which I'll put up a PR for.

File: src/app/zap-templates/templates/app/helper.js [hunk 1]
```diff
@@ -394,21 +394,39 @@ async function zapTypeToClusterObjectType(type, isDecodable, options)
...
-    if (await typeChecker('isBitmap')) {
+    if (types.isBitmap) {
       return 'chip::BitFlags<' + ns + type + '>';
     }
 
-    if (await typeChecker('isStruct')) {
-      passByReference = true;
-      return ns + 'Structs::' + type + '::' + (isDecodable ? 'DecodableType' : 'Type');
+    if (types.isEvent) {
+      return ns + 'Events::' + type + '::' + (isDecodable ? 'DecodableType' : 'Type');
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/templates/app/helper.js:424]: This should have `passByReference = true`, no?

**Review by bzbarsky-apple — APPROVED** [review]
I'd leave the event named SoftwareFault, rename the struct to SoftwareFaultStruct for now, and in a followup we can remove the struct.

File: src/app/zap-templates/templates/app/helper.js (hunk 1 above)
Comment (vivien-apple) [reply src/app/zap-templates/templates/app/helper.js]: Yes...

**Comment by woody-apple** [comment]
Fast tracking, given > 3 days for review.

**Pull request merged**